### ISSUE TITLE
学习日志：2017-12

### ISSUE THREAD

**Issue by JunningWu** [JunningWu/AIChip#5]
2017-12-14

论文学习：
paper《In-Datacenter Performance Analysis of a Tensor Processing Unit-2017》
专利《Prefetching Weights for a Neural Network Processor》

NVDLAonFPGA
1.安装完成Vivado2017.1，windows版本加入source代码，仿真出错，正在调试；Linux版本借助于脚本，可以进行仿真
2.重新学习vivado的设计流程，文档《Vivado Design Suite Tutorial: Design Flows Overview》，

NVDLA 代码学习
使用VCS工具，可以将NVDLA所有的测试例进行仿真，学习NVDLA的架构和代码设计

NVDLA模拟器VP
安装完成VP，并使用提供的Linux镜像文件可以boot起来，并进行了hello world和BDMA测试例的仿真。

NVDLA软件SW Stack
还未进行，了解了一下umd和kmd的说明文档。

**Comment by JunningWu** [comment]
2017-12-15
论文学习：
专利 《Batch Processing in a Neural Network Processor》

NVDLAonFPGA


NVDLA 代码学习


NVDLA模拟器VP


NVDLA软件SW Stack

**Comment by JunningWu** [comment]
**2017-12-25**

**论文学习：**

《ImageNet Classification with Deep Convolutional Neural Networks》学习AlexNet，


**NVDLAonFPGA**

**NVDLA 代码学习**
**01.** 
NVDLA的测试例cc_alexnet_conv5_relu5_int16_dtest_cvsram，实现的是AlexNet的第五层卷积层和ReLU层，input为13x13x384，权值为3x3x384x256。所使用的功能部件为SDP，CDMA，CSC，CMACA，CMACB，CACC等。主要完成卷积操作和ReLU。
![image](https://user-images.githubusercontent.com/4930405/34331046-4f7a406a-e95e-11e7-99b1-954f78b1131a.png)

**NVDLA模拟器VP**


**NVDLA软件SW Stack**
重新学习了一遍NVDLA的软件工具链，并做了总结ppt。
![image](https://user-images.githubusercontent.com/4930405/34331127-e5c9329c-e95e-11e7-8583-66291e4562f5.png)

**Comment by JunningWu** [comment]
**2017-12-27**

**论文学习：**

**NVDLAonFPGA**

**NVDLA 代码学习**

**NVDLA模拟器VP**

在docker模式下，使用NVDLA提供的loadable文件，可以顺利执行nvdla_runtime。
然而，当使用nvdla_compiler编译Caffe Model [Zoo提供的alexnet网络，可以顺利编译，但是在执行的时候，会报错，应该是与kernel相关的，错误，具体问题描述，已经提交到NVDLA的issue列表。 
https://github.com/nvdla/vp/issues/1

**NVDLA软件SW Stack**

通过测试，知道了nvdla_compiler和nvdla_runtime更多的执行选项。
Usage: ./nvdla_runtime [-options] --loadable <loadable_file>
where options include:
-h print this help message
-s launch test in server mode
--loadable <loadable_file>
--image <image_file>
--imgshift <shift_value>
--imgscale <scale_value>
--imgpower <power_value>
--softmax

具体进度如上述描述。


**Caffe**

在虚拟机上搭建了caffe的开发环境，顺利install了caffe，并在juyper notebook上面完成了MNIST的tutorial。还在进一步摸索过程中。

**Comment by Andrawzyf** [comment]
vcs这个软件资源网上好像很难找到。。。

**Comment by JunningWu** [comment]
@Andrawzyf 单位的资源
